api/middleware/applicationMiddleware.js

```diff
@@ -86,11 +86,12 @@ const checkApplicationExists = async (req, res, next) => {
 
 const checkRole = async (req, res, next) => {
   const application = req.application;
+
   try {
     if (application.position === 3) {
       const mentorData = await getMentorIntake(application.profile_id);
       if (!mentorData) {
-        return next({
+        next({
```

Comment:
I did not know this about next --> "if you call next with an object inside, the last function in your middleware pipeline will be called. In this case, that function is our global error handling function." That makes sense and seems like the return would be redundant.

Great catch and insightful!